src/Form/Type/AddProductsToCartType.php

```diff
@@ -46,32 +49,26 @@ public function __construct(
 
     public function buildForm(FormBuilderInterface $builder, array $options): void
     {
-        /** @var WishlistProductInterface $wishlistProduct */
-        foreach ($options['wishlist_products'] as $key => $wishlistProduct) {
-            $builder
-                ->add($key, AddToCartType::class, [
-                    'label' => false,
-                    'required' => false,
-                    'product' => $wishlistProduct->getProduct(),
-                    'data' => $this->addToCartCommandFactory->createWithCartAndCartItem(
-                        $options['cart'],
-                        $this->createCartItem($wishlistProduct)
-                    ),
-                    'is_wishlist' => true,
-                ])
-            ;
-        }
-    }
+        $builder->addEventListener(FormEvents::POST_SET_DATA, function (FormEvent $event) use ($options): void {
+            $form = $event->getForm();
 
-    public function configureOptions(OptionsResolver $resolver): void
-    {
-        $resolver
-            ->setRequired('cart')
-            ->setRequired('wishlist_products')
-            ->setAllowedTypes('wishlist_products', Collection::class)
-            ->setDefault('data_class', null)
-            ->setDefault('validation_groups', $this->validationGroups)
-        ;
+            $wishlistProduct = $form->getData()->getWishlistProduct();
+
+            $form
+                ->add('cartItem', AddToCartType::class, [
+                'label' => false,
+                'required' => false,
+                'product' => $wishlistProduct->getProduct(),
+                'data' => $this->addToCartCommandFactory->createWithCartAndCartItem(
+                    $options['cart'],
+                    $this->createCartItem($wishlistProduct)
+                ),
+                'is_wishlist' => true,
```

Comment:
What's the purpose of this flag?

Reply:
It is set to avoid displaying add to wishlist button